DOC Added example comparing L1-based models to ARD user guide

supersom opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs
Towards #30621 
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


#### What does this implement/fix? Explain your changes.
* Added example comparing L1-based models to ARD user guide
* Integrated (existing) example comparing Bayesian Regressors into text

#### Any other comments?
If integration into text is the preferred model, should I remove the 'example' section for ARD?


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
https://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

supersom committed:
DOC Added example comparing L1-based models to ARD user guide and integrated example comparing Bayesian Regressors into text

supersom committed:
Added example comparing L1-based models to ARD user guide and integrated example comparing Bayesian Regressors into text

supersom committed:
Merge branch 'main' into doc_link_example

supersom committed:
Merge branch 'main' into doc_link_example

StefanieSenger reviewed (commented):
Thank you for taking this back up, @supersom!

> If integration into text is the preferred model, should I remove the 'example' section for ARD?

Yes, you definitely should remove the examples section in `doc/modules/linear_model.rst` (li. 845-849 currently), since this contains one of the two examples you have linked above and we don't want to have duplicated links.


I would like to finish on these two links, since it seems that #30587 is stalled.

Apart from this, `sphx_glr_auto_examples_linear_model_plot_lasso_and_elasticnet.py` is already linked sufficiently in the user guide, I belive, and `sphx_glr_auto_examples_linear_model_plot_ard.py` is already linked in ARDRegressor's docstring. Is there something from your perspective that, were one of the two links need to be linked?

supersom committed:
DOC removing 'Examples' section from ARD user guide - examples inlined

supersom committed:
DOC Added example comparing L1-based models to Lasso, ElasticNet and ARD docstrings and demonstrating ARD to ARD docstring [as in PR #30587]

supersom committed:
Merge branch 'main' into doc_link_example

supersom committed:
Merge branch 'main' into doc_link_example

supersom committed:
Merge branch 'main' of github.com:scikit-learn/scikit-learn into doc_link_example

supersom committed:
Merge branch 'doc_link_example' of github.com:supersom/scikit-learn into doc_link_example

supersom commented:
Thanks for clarifying, @StefanieSenger!
I have removed the examples section in `doc/modules/linear_model.rst` and added the changes from #30587.

> Is there something from your perspective that, were one of the two links need to be linked?

I'm not sure I followed you on this one - could you help me out please?

supersom committed:
Merge branch 'main' into doc_link_example

supersom committed:
DOC FIX docstring list indents

supersom committed:
Merge branch 'doc_link_example' of github.com:supersom/scikit-learn into doc_link_example

StefanieSenger commented:
Thank you for your further work, @supersom.

> I'm not sure I followed you on this one - could you help me out please?

Oh I see I did not express myself well, sorry about that. I meant that I don't think the links from the other PR are really necessary, and asked for your opinion.

According to CI there are also several places with double line breaks (we are using [numpydoc style guide](https://numpydoc.readthedocs.io/en/latest/format.html)):
`- GL03: Double line break found; please use only one blank line to separate sections or paragraphs, and do not leave blank lines at the end of docstrings`

Could you please fix these?

supersom committed:
DOC FIX removing double line breaks

supersom committed:
DOC FIX adding line break above list items

supersom committed:
Merge branch 'main' into doc_link_example

StefanieSenger reviewed (commented):
Thanks for fixing the rendering issues, @supersom.

Now, two examples appear twice, and we need to remove the duplicates.

I am not so convinced about the links in the docstrings, since I think general examples like these are more a matter for the user guide, and we want to keep examples section in the API reference as clean as possible and they should be for pure code examples only.

But I will leave this for a maintainer to decide on.

supersom committed:
DOC FIX removing duplicate links

supersom committed:
Merge branch 'doc_link_example' of github.com:supersom/scikit-learn into doc_link_example

supersom commented on doc/modules/linear_model.rst:
```diff
@@ -837,13 +837,15 @@ prior over all :math:`\lambda_i` is chosen to be the same gamma distribution
+See :ref:`sphx_glr_auto_examples_linear_model_plot_ard.py` for a worked-out comparison between ARD and `Bayesian Ridge Regression`_.
 
+See :ref:`sphx_glr_auto_examples_linear_model_plot_lasso_and_elasticnet.py` for a comparison between various methods - Lasso, ARD and ElasticNet - on correlated data.
 
```
> Now these two are linked twice.

Thank you so much for pointing that out, @StefanieSenger - I overlooked auto-merge getting too helpful! Should be fixed now.

> we want to keep examples section in the API reference as clean as possible and they should be for pure code examples only.

That sounds like a reasonable plan - code examples in docstrings and general examples in user guides. Since #30587 was already underway, I just brought those changes over TBH.

supersom committed:
Merge branch 'main' into doc_link_example

StefanieSenger reviewed (commented):
Thanks for your most recent adjustments, @supersom.

Now everything is in good shape. Only the question on whether to add these examples to the API reference of `ElasticNet`, `Lasso` and `ARDRegressor` is still open. What would you think, @adrinjalali?

supersom commented:
> Thanks for your most recent adjustments, @supersom.
> 
> Now everything is in good shape. Only the question on whether to add these examples to the API reference of `ElasticNet`, `Lasso` and `ARDRegressor` is still open. What would you think, @adrinjalali?

Hi @adrinjalali , @StefanieSenger  - Please let me know how we could take this to a merge.

arpanmukherjee38 commented:
Hi! I'm new to contributing and would like to work on this issue. Can I take it?
